tests/pseudoterminal.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+const std::string INVALID_PORT = "invalid";
+
+namespace pseudoterminal
+{
+    class ConnectedPorts
```

Comment:
maybe change the name to pseudoterminal and remove the namespace.

Reply:
I think this is the most natural approach. I can see a situation where logic related to pseudoterminal has to be implemented to support the setup, yet is not directly related to lifecycle of an instance of "connected ports"

Reply:
Good point. Then I would rename the class to pseudoPort and add a terminal class that implements the connection between the ports. This way you can also future proof for multiple port connections on one terminal.